qa-tests-backend/src/main/groovy/orchestratormanager/Kubernetes.groovy

```diff
@@ -877,21 +889,44 @@ class Kubernetes implements OrchestratorMain {
 
     Set<String> getStaticPodCount(String ns = null) {
         return evaluateWithRetry(2, 3) {
-            // This method assumes that a static pod name will contain the node name that the pod is running on
-            def nodeNames = client.nodes().list().items.collect { it.metadata.name }
+            // This method assumes that a static pod will either have no OwnerReferences,
+            // it will have a owner not tracked by ACS, or it will be a kube-proxy pod
             Set<String> staticPods = [] as Set
             PodList podList = ns == null ? client.pods().list() : client.pods().inNamespace(ns).list()
             podList.items.each {
-                for (String node : nodeNames) {
-                    if (it.metadata.name.contains(node)) {
-                        staticPods.add(it.metadata.name[0..it.metadata.name.indexOf(node) - 2])
-                    }
+                if (!isRunning(it)) {
+                    return
+                }
+                if (it.getMetadata().getOwnerReferences().size() > 0) {
+                        if (!isKubeProxyPod(it) && ownerIsTracked(it.getMetadata())) {
+                            return
+                        }
                 }
```

Comment:
Would `if (A && !B && C)` work here as well?

Reply:
It does work but I divided the ifs on purpose I think it reads better. I don't have a strong opinion though if you want to I can join them again 🙂

Reply:
No need, you can leave it like it is.